src/System.CommandLine/CommandLineConfiguration.cs

```diff
@@ -56,10 +56,10 @@ public class CommandLineConfiguration
             _helpBuilderFactory = helpBuilderFactory;
         }
 
-        private static HelpBuilder DefaultHelpBuilderFactory(BindingContext context)
+        internal static HelpBuilder DefaultHelpBuilderFactory(BindingContext context)
         {
             int maxWidth = int.MaxValue;
-            if (context.Console is SystemConsole systemConsole)
```

Comment:
@jonsequitur Limitation for testability is here - getting width was limited to `SystemConsole` which works only with `Console`.

Reply:
Agreed. The `TestConsole` deliberately doesn't have a width concept. It's intended only to represent standard streams. This approach was intended to isolate an implementation detail that hopefully will go away entirely when the standard streams and terminal concepts are more decoupled and testable lower down the stack. For that reason, passing a width via the `TestConsole` is a design direction we avoided from early on.  

I'd prefer to find a different way to pass this parameter in that makes it clearer at the public API level where it comes from.

Reply:
Changed it to be `CommandLineBuilder` decorator parameter instead.